source/slang/slang-check-decl.cpp

```diff
@@ -1485,6 +1702,14 @@ IntVal* SemanticsVisitor::_validateCircularVarDefinition(VarDeclBase* varDecl)
         nullptr);
 }
 
+void SemanticsDeclModifiersVisitor::visitInterfaceDecl(InterfaceDecl* interfaceDecl)
+{
+    visitDecl(interfaceDecl);
+
+    if (interfaceDecl->hasModifier<AnyValueSizeAttribute>())
```

Comment:
Check all the `dyn` interface restrictions here, also make sure to check that `dyn` interface itself cannot be a generic.